app/models/file.ts

```diff
@@ -150,15 +151,16 @@ export default class FileModel extends BaseFileItem {
                 resource: node.id,
                 ...options,
             }),
-            success: (data, _, xhr) => {
-                data.status = xhr.status;
+            success: (_, __, xhr) => {
+                status = xhr.status;
             },
         });
-        return req;
+        return { requestData, status };
```

Comment:
I wonder if it would make more sense to ensure that requestData is an object here rather than later down the line.